core/gui/src/app/workspace/service/editor-state/editor-state.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root",
+})
+export class EditorStateService {
+  private editorStates: Map<string, BehaviorSubject<boolean>> = new Map();
```

Comment:
You don't have to make a separate states service since there is already a codeEditor service. You can merge them.